### ISSUE TITLE
isinstance check on object fails to narrow generic bounded type

### ISSUE THREAD

**Issue by sid-kap** [astral-sh/ty#2793]
### Summary

This fails type checking:
```py
class A[T: int]:
    t: T

def foo(x: int) -> None:
    pass

def bar(x: object) -> None:
    assert isinstance(x, A)
    foo(x.t)
```
with the error
```
Argument to function `foo` is incorrect: Expected `int`, found `object` (invalid-argument-type) [Ln 9, Col 9]
```
https://play.ty.dev/11c910b7-e060-4c9f-9a8c-1e41b7937619

I believe this should not error.

Apologies if this is a duplicate! I see many other issues with the narrowing label that seem very similar (#653, #1503, #2726, #2782)

### Version

ty 0.0.15 (eb43dff06 2026-02-04)

**Comment by AlexWaygood** [comment]
Thanks for the report! I think this is #1109
